app/domain/authentication/authn_jwt/identity_providers/identity_from_decoded_token_provider.rb

```diff
@@ -9,7 +9,7 @@ def initialize(authentication_parameters)
           @secret_fetcher = Conjur::FetchRequiredSecrets.new
           @resource_class = ::Resource
```

Comment:
these 2 should go in the method signature with default values so they can me mocked in UTs. please see that you do this for dependencies in all regular classes.